### PR TITLE
FIX FOR: AS-39 clicking on topic or source from the autocomplete returns no results 

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

**Comment by bbfsdev** [comment]
I have played with your branch, your solution brings up the previous bug: https://issues.kbb1.com/issue/AR-234

To recreate:
Go to programs page and filter by some program, for example:
https://archive.kbb1.com/he/programs?program=%7CJRR182e6
Now search for topic or source:
You will have both filters applied, which is wrong, i.e., you copy the filter from the previous page. You should only copy the search filter, i.e., the source or topic filter.

See pictures attached in email...

**Comment by YuriHechter** [comment]
fixed

File: src/components/Search/OmniBox.js [hunk 1]
```diff
@@ -126,12 +124,18 @@ export class OmniBox extends Component {
       this.doSearch(data.result.title);
     } else if (category === 'tags') {
       this.props.updateQuery('');
-      this.props.addFilterValue('search', 'topics-filter', this.props.getTagPath(data.result.key).map(p => p.id));
-      this.doSearch('');
+      const path = this.props.getTagPath(data.result.key).map(p => p.id)
+      const queryValue = path.join('_');
+      const queryKey = "topic=";
+      this.props.addFilterValue('search', 'topics-filter', path);
+      this.doSearch('', queryKey.concat(queryValue));
```
Comment (bbfsdev) [inline src/components/Search/OmniBox.js]: This approach is good, but you have to use built it functions to build the query:
See this https://github.com/Bnei-Baruch/kmedia-mdb/blob/master/src/filters/definitions/topicsFilter.js#L6

File: src/components/Search/OmniBox.js [hunk 2]
```diff
@@ -126,12 +124,18 @@ export class OmniBox extends Component {
...
     } else if (category === 'sources') {
       this.props.updateQuery('');
-      this.props.setFilterValue('search', 'sources-filter', this.props.getSourcePath(data.result.key).map(p => p.id));
-      this.doSearch('');
+      const path = this.props.getSourcePath(data.result.key).map(p => p.id)
+      const queryValue = path.join('_');
+      const queryKey = "source=";
+      this.props.setFilterValue('search', 'sources-filter', path);
```
Comment (bbfsdev) [inline src/components/Search/OmniBox.js]: This approach is good, but you have to use built it functions to build the query:
See this https://github.com/Bnei-Baruch/kmedia-mdb/blob/master/src/filters/definitions/sourcesFilter.js#L6

**Comment by bbfsdev** [comment]
The approach is good, just fix the two comments and you are ok.
Make sure everything works.

File: src/components/Search/OmniBox.js (hunk 1 above)
Comment (YuriHechter) [reply src/components/Search/OmniBox.js]: done

**Comment by bbfsdev** [comment]
Excellent.

**Comment by bbfsdev** [comment]
Please pull from master and sent to Edo

**Pull request merged**